src/test/java/org/assertj/core/api/object/ObjectAssert_extracting_Test.java

```diff
@@ -67,17 +75,37 @@ public void should_use_property_field_names_as_description_when_extracting_tuple
   }
 
   @Test
-  public void should_keep_existing_description_if_set_when_extracting_tuples_list() {
+  public void should_keep_existing_description_if_set_when_extracting_single_property() {
     Employee luke = new Employee(2L, new Name("Luke", "Skywalker"), 26);
 
     assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> assertThat(luke).as("check luke first name")
                                                                                      .extracting("name.first")
-                                                                                     .isEmpty())
+                                                                                     .isNull())
                                                    .withMessageContaining("[check luke first name]");
   }
 
   @Test
-  public void should_allow_to_specify_type_comparator_after_using_extracting_on_object() {
+  public void should_keep_existing_description_if_set_when_extracting_tuples_list() {
+    Employee luke = new Employee(2L, new Name("Luke", "Skywalker"), 26);
+
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> assertThat(luke).as("check luke first and last name")
+                                                                                     .extracting("name.first", "name.last")
+                                                                                     .isEmpty())
+                                                   .withMessageContaining("[check luke first and last name]");
+  }
+
+  @Ignore
+  public void should_allow_to_specify_type_comparator_after_using_extracting_with_single_parameter_on_object() {
```

Comment:
I created this test to have the same use case as the existing one with multiple parameters, but from my understanding there is no way to set a comparator for an comparison between objects. Am I missing something?
It could probably work with some more intrusive changes, maybe in `org.assertj.core.internal.Objects`.

Reply:
After some investigation, I think the main "issue" is that the `extracting(String)`/`extracting(String[])` methods are returning an `ObjectAssert` instead of the specific type assertion. I have the feeling some improvement could be done in `PropertySupport` and `FieldSupport`, but I would prefer to do it in a separate pull request.

Would that be OK for you? Shall I leave this test as ignored or do you prefer to remove it?

Reply:
`usingComparatorForType` is only used in the context of field by field comparison.
To use a specific comparator with `isEqualTo`, one had to call `public SELF usingComparator(Comparator<? super ACTUAL> customComparator)` but in our case ACTUAL = Object and thus it is not possible to use a `BigDecimal` comparator.

To allow this use case, we would need to allow users to specify the type of the extracted value + changing the signature of extracting to:
```
public <T> AbstractObjectAssert<?, T> extracting(String propertyOrField, Class<T> extractedType)
```
and a bunch of other changes to cascade the introduction of the generic parameter T.

That is I think valuable but definitely in another pull request.
```
T value = byName(propertyOrField).apply(actual);

Reply:
I "fixed" the test by using an Object comparator (hacky but possible):
```
    Comparator<Object> heightComparator = (o1, o2) -> {
      if (o1 instanceof BigDecimal) return BIG_DECIMAL_COMPARATOR.compare((BigDecimal) o1, (BigDecimal) o2);
      throw new IllegalStateException("only supported for BigDecimal");
    };
    assertThat(obiwan).extracting(Person::getHeight)
                      .usingComparator(heightComparator)
                      .isEqualTo(new BigDecimal("1.82"));
```

Reply:
oh and to ignore a test in JUnit 5, you must use [`@Disabled`](https://junit.org/junit5/docs/5.0.0-M2/api/org/junit/jupiter/api/Disabled.html), `@Ignore` is for JUnit 4 ;-)

Reply:
Nice hack :-) code updated!